src/ImageBuilder/test/Update-AzImageBuilderTemplate.Tests.ps1

```diff
@@ -0,0 +1,24 @@
+if(($null -eq $TestName) -or ($TestName -contains 'Update-AzImageBuilderTemplate'))
+{
+  $loadEnvPath = Join-Path $PSScriptRoot 'loadEnv.ps1'
+  if (-Not (Test-Path -Path $loadEnvPath)) {
+      $loadEnvPath = Join-Path $PSScriptRoot '..\loadEnv.ps1'
+  }
+  . ($loadEnvPath)
+  $TestRecordingFile = Join-Path $PSScriptRoot 'Update-AzImageBuilderTemplate.Recording.json'
+  $currentPath = $PSScriptRoot
+  while(-not $mockingPath) {
+      $mockingPath = Get-ChildItem -Path $currentPath -Recurse -Include 'HttpPipelineMocking.ps1' -File
+      $currentPath = Split-Path -Path $currentPath -Parent
+  }
+  . ($mockingPath | Select-Object -First 1).FullName
+}
+
+Describe 'Update-AzImageBuilderTemplate' {
+    It 'UpdateExpanded' {
+        {
+            $template = Update-AzImageBuilderTemplate -Name $env.newTemplateName3 -ResourceGroupName $env.rg -Tag @{"123"="abc"}
```

Comment:
- to verify whether updated successfully, you need to perform update and get, and compare the input parameters with response of get
- if you want to test whether tag was updated, please use line 21 to compare the actual tag in get response with the input tag. check the template name is meaningless here

Reply:
After the Update-Az command runs successfully on the server side, it encapsulates the updated content as a new object and returns it locally, so in the unit test I check that the returned result exists and that the returned content is correct should prove that the command ran correctly.

Reply:
- response of PATCH was not guaranteed to represent the state of the current resource but GET does
- you are updating the tags, how do you know the returned content is correct without verifying it?

Reply:
Done,